apiClient/client.ts

```diff
@@ -207,3 +208,8 @@ export async function getUserDetails(
     return new LocalError(e.message, ENDPOINT_UNAVAILABLE)
   }
 }
+
+export const isGenericError = (x: GenericApiError) => {
```

Comment:
You can define the function like this:

```ts
export const isGenericError = (
  x: Record<string, unknown>
): x is GenericApiError => {
  const { statusCode } = x
  return typeof statusCode === 'number' && statusCode > 400
}
```

Then it'll avoid the need to do `as API.GenericApiError` when passing messages into `API.isGenericError`. (As an FYI, `Record<string, unknown>` is generally equivalent to `object`)

Reply:
Nice, the `x is GenericApiError` is new to me -- I tried the Record in an earlier permutation.